core/src/main/kotlin/Model/SourceLinks.kt

```diff
@@ -6,31 +6,61 @@ import com.intellij.psi.PsiNameIdentifierOwner
 import org.jetbrains.dokka.DokkaConfiguration.SourceLinkDefinition
 import org.jetbrains.kotlin.psi.psiUtil.startOffset
 import java.io.File
+import java.io.IOException
 
 
 fun DocumentationNode.appendSourceLink(psi: PsiElement?, sourceLinks: List<SourceLinkDefinition>) {
     val path = psi?.containingFile?.virtualFile?.path ?: return
 
     val target = if (psi is PsiNameIdentifierOwner) psi.nameIdentifier else psi
-    val absPath = File(path).absolutePath
-    val linkDef = sourceLinks.firstOrNull { absPath.startsWith(it.path) }
-    if (linkDef != null) {
-        var url = linkDef.url + path.substring(linkDef.path.length)
-        if (linkDef.lineSuffix != null) {
+    val pair = determineSourceLinkDefinition(path, sourceLinks)
+    if (pair != null) {
+        val (sourceLinkDefinition, sourceLinkAbsPath) = pair
+        var url = determineUrl(path, sourceLinkDefinition, sourceLinkAbsPath)
+        if (sourceLinkDefinition.lineSuffix != null) {
             val line = target?.lineNumber()
             if (line != null) {
-                url += linkDef.lineSuffix + line.toString()
+                url += sourceLinkDefinition.lineSuffix + line.toString()
             }
         }
-        append(DocumentationNode(url, Content.Empty, NodeKind.SourceUrl),
-                RefKind.Detail);
+        append(DocumentationNode(url, Content.Empty, NodeKind.SourceUrl), RefKind.Detail)
     }
 
     if (target != null) {
         append(DocumentationNode(target.sourcePosition(), Content.Empty, NodeKind.SourcePosition), RefKind.Detail)
     }
 }
 
+private fun determineSourceLinkDefinition(
+    path: String,
+    sourceLinks: List<SourceLinkDefinition>
+): Pair<SourceLinkDefinition, String>? {
+    val absPath = File(path).absolutePath
+    return sourceLinks
+        .asSequence()
+        .map {
+            val file = File(it.path)
+            val sourceLinkAbsPath = try {
+                file.canonicalPath
+            } catch (e: IOException) {
+                //well then, it's probably messed up anyway.
+                file.absolutePath
+            }
+            it to sourceLinkAbsPath
+        }.firstOrNull { (_, sourceLinkAbsPath) ->
+            absPath.startsWith(sourceLinkAbsPath)
+        }
+}
+
+private fun determineUrl(
+    path: String,
+    sourceLinkDefinition: SourceLinkDefinition,
+    sourceLinkAbsPath: String
+): String {
+    val relativePath = path.substring(sourceLinkAbsPath.length)
```

Comment:
There should be better methods to determine relative path, like `File.relativeTo`

Reply:
A bit overkill here IMO. We already checked that both paths have the same root. We don't need to do this again with `relativeTo`